frontend/src/metabase/containers/CollectionMoveModal.tsx

```diff
@@ -0,0 +1,93 @@
+import { useState } from "react";
+
+import { t } from "ttag";
+
+import Button from "metabase/core/components/Button";
+import ModalContent from "metabase/components/ModalContent";
+
+import CollectionPicker from "metabase/containers/CollectionPicker";
+import CreateCollectionModal from "metabase/collections/containers/CreateCollectionModal";
+import { Collection, CollectionId } from "metabase-types/api";
+import { ButtonContainer } from "./CollectionMoveModal.styled";
+
+interface CollectionMoveModalProps {
+  title: string;
+  onClose: () => void;
+  onMove: (collection: any) => void;
+  initialCollectionId?: number | string | null;
+}
+
+export const CollectionMoveModal = ({
+  title,
+  onClose,
+  onMove,
+  initialCollectionId,
+}: CollectionMoveModalProps) => {
+  // will eventually be the collection object representing the selected collection
+  // we store the whole object instead of just the ID so that we can use its
+  // name in the action button, and other properties
+  //
+  //  undefined = no selection
+  //  null = root collection
+  //  number = non-root collection id
+  //
+  const [selectedCollectionId, setSelectedCollectionId] =
+    useState(initialCollectionId);
+
+  // whether the move action has started
```

Comment:
I think we can delete this comment and the one above about the id eventually being the collection object, while keeping the one explaining what each type of id represents.

It's been five years since they were originally written, so I think by this point we are set on using the id and not having a moving/error state. If someone really wants these changes they can write a product doc or file an issue.